go/proxyd/backend.go

```diff
@@ -1,229 +1,246 @@
 package proxyd
 
 import (
-  "bytes"
-  "encoding/json"
-  "errors"
-  "fmt"
-  "github.com/ethereum/go-ethereum/log"
-  "github.com/prometheus/client_golang/prometheus"
-  "github.com/prometheus/client_golang/prometheus/promauto"
-  "io"
-  "io/ioutil"
-  "math"
-  "math/rand"
-  "net/http"
-  "sync/atomic"
-  "time"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"io"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net/http"
+	"sync/atomic"
+	"time"
 )
 
 const (
-  JSONRPCVersion = "2.0"
+	JSONRPCVersion = "2.0"
 )
 
 var (
-  ErrNoBackend            = errors.New("no backend available for method")
-  ErrBackendsInconsistent = errors.New("backends inconsistent, try again")
-  ErrBackendOffline       = errors.New("backend offline")
-
-  backendRequestsCtr = promauto.NewCounterVec(prometheus.CounterOpts{
-    Namespace: "proxyd",
-    Name:      "backend_requests_total",
-    Help:      "Count of backend requests.",
-  }, []string{
-    "name",
-  })
-
-  backendErrorsCtr = promauto.NewCounterVec(prometheus.CounterOpts{
-    Namespace: "proxyd",
-    Name:      "backend_errors_total",
-    Help:      "Count of backend errors.",
-  }, []string{
-    "name",
-  })
-
-  backendPermanentErrorsCtr = promauto.NewCounterVec(prometheus.CounterOpts{
-    Namespace: "proxyd",
-    Name:      "backend_permanent_errors_total",
-    Help:      "Count of backend errors that mark a backend as offline.",
-  }, []string{
-    "name",
-  })
+	ErrNoBackend            = errors.New("no backend available for method")
+	ErrBackendsInconsistent = errors.New("backends inconsistent, try again")
+	ErrBackendOffline       = errors.New("backend offline")
+
+	backendRequestsCtr = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "proxyd",
+		Name:      "backend_requests_total",
+		Help:      "Count of backend requests.",
+	}, []string{
+		"backend_name",
```

Comment:
We should add the method label to request count and errors.

Reply:
Yep.